### PR TITLE
chore(sql): log exception call stack on query failures

### PR DESCRIPTION
**Pull request by ideoma**
When an unexpected exception (not SqlException/CairoException) happens during query execution it is not well logged

```
2024-10-08T10:19:35.722700Z E i.q.g.e.QueryProgress err [id=4, sql=`select k, count_distinct(l) from x order by k`, principal=admin, cache=false, jit=false, time=671733775, msg=null, errno=0, pos=0]
```

The error log misses all the exception details. This PR turns the query error logging into:

```
2024-10-08T13:26:44.071977Z E i.q.g.e.QueryProgress err [id=4, sql=`select k, count_distinct(l) from x order by k`, principal=admin, cache=false, jit=false, time=545100842, exception=
java.lang.NullPointerException: Cannot invoke "io.questdb.griffin.engine.groupby.GroupByAllocator.malloc(long)" because "this.allocator" is null
	at io.questdb.griffin.engine.groupby.GroupByLongHashSet.rehash(GroupByLongHashSet.java:199)
	at io.questdb.griffin.engine.groupby.GroupByLongHashSet.merge(GroupByLongHashSet.java:125)
	at io.questdb.griffin.engine.functions.groupby.CountDistinctLongGroupByFunction.merge(CountDistinctLongGroupByFunction.java:146)
	at io.questdb.cairo.map.OrderedMap.mergeVarSizeKey(OrderedMap.java:474)
	at io.questdb.cairo.map.OrderedMap.merge(OrderedMap.java:304)
	at io.questdb.griffin.engine.table.AsyncGroupByAtom.mergeOwnerMap(AsyncGroupByAtom.java:394)
	at io.questdb.griffin.engine.table.AsyncGroupByRecordCursor.buildMap(AsyncGroupByRecordCursor.java:223)
	at io.questdb.griffin.engine.table.AsyncGroupByRecordCursor.hasNext(AsyncGroupByRecordCursor.java:137)
	at io.questdb.griffin.engine.orderby.SortedLightRecordCursor.buildChain(SortedLightRecordCursor.java:123)
	at io.questdb.griffin.engine.orderby.SortedLightRecordCursor.hasNext(SortedLightRecordCursor.java:79)
	at io.questdb.griffin.engine.QueryProgress$RegisteredRecordCursor.hasNext(QueryProgress.java:292)
	at io.questdb.cutlass.pgwire.PGConnectionContext.sendCursor0(PGConnectionContext.java:2691)
	at io.questdb.cutlass.pgwire.PGConnectionContext.sendCursor(PGConnectionContext.java:2681)
	at io.questdb.cutlass.pgwire.PGConnectionContext.processExecute(PGConnectionContext.java:2390)
	at io.questdb.cutlass.pgwire.PGConnectionContext.processExec(PGConnectionContext.java:2382)
	at io.questdb.cutlass.pgwire.PGConnectionContext.parse(PGConnectionContext.java:1878)
	at io.questdb.cutlass.pgwire.PGConnectionContext.handleClientOperation(PGConnectionContext.java:511)
	at io.questdb.cutlass.pgwire.PGWireServer$1.lambda$$0(PGWireServer.java:96)
	at io.questdb.network.AbstractIODispatcher.processIOQueue(AbstractIODispatcher.java:199)
	at io.questdb.cutlass.pgwire.PGWireServer$1.run(PGWireServer.java:126)
	at io.questdb.mp.Worker.run(Worker.java:151)
```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 25 / 25 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/QueryProgress.java|25|25|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/QueryProgress.java [hunk 1]
```diff
@@ -82,20 +82,32 @@ public static void logError(
...
+            final int errno = e instanceof CairoException ? ((CairoException) e).getErrno() : 0;
+            LOG.errorW()
+                    .$("err")
+                    .$(" [id=").$(sqlId)
+                    .$(", sql=`").utf8(sqlText).$('`')
+                    .$(", principal=").$(executionContext.getSecurityContext().getPrincipal())
+                    .$(", cache=").$(executionContext.isCacheHit())
+                    .$(", jit=").$(jit)
+                    .$(", time=").$(executionContext.getCairoEngine().getConfiguration().getNanosecondClock().getTicks() - beginNanos)
+                    .$(", msg=").$(e.getMessage())
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/griffin/engine/QueryProgress.java:96]: This should use `getFlyweightMessage()`

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 31 / 33 (93.94%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/QueryProgress.java|31|33|93.94%|

File: core/src/main/java/io/questdb/griffin/engine/QueryProgress.java (hunk 1 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/griffin/engine/QueryProgress.java:103]:
```suggestion
                    .$(", msg=").$(e.getFlyweightMessage())
```

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 34 (94.12%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/QueryProgress.java|32|34|94.12%|

**Review by puzpuzpuz — APPROVED** [review]

**Pull request merged**